main/src/main/scala/sbt/Defaults.scala

```diff
@@ -464,13 +464,12 @@ object Defaults extends BuildCommon {
     },
   )
 
-  private[sbt] def toAbsoluteSourceMapper(fc: FileConverter): Position => Option[Position] = {
-    pos =>
-      val newPath: Optional[String] = pos.sourcePath
-        .map { id =>
-          fc.toPath(VirtualFileRef.of(id)).toAbsolutePath.toString
-        }
-      Some(
+  private[sbt] def toAbsoluteSourceMapper(pos: Position): Option[Position] = {
+    val newPath: Option[String] =
+      pos.sourceFile.asScala.map(file => file.getAbsolutePath)
```

Comment:
I'd avoid the use of `sourceFile` and instead check if id contains `<` or something.

Reply:
`<` is allowed in Unix file names.
So I would rather use a `isValid(path: String): Boolean` that tries to create a `java.nio.Path` and returns `false` in case of failure (`InvalidPathException`).